pymongo/monitoring.py

```diff
@@ -1362,9 +1364,14 @@ def publish_command_success(self, duration, reply, command_name,
             command was sent to.
           - `op_id`: The (optional) operation id for this operation.
           - `service_id`: The service_id this command was sent to, or ``None``.
+          - `speculative_hello`: Was the command sent with speculative auth?
         """
         if op_id is None:
             op_id = request_id
+        if speculative_hello:
+            # Redact entire response when the command started contained
+            # speculativeAuthenticate.
+            reply = {}
```

Comment:
To understand why we need to do this see this new test: https://github.com/mongodb/specifications/blob/289f405/source/command-monitoring/tests/unified/redacted-commands.yml#L232-L242